### PR TITLE
Further refactoring of `pd_gmres()` + test renaming

### PR DESCRIPTION
**Pull request by jhabriel**
This PR introduces further refactoring of the `pd_gmres` function. The following changes have been introduced:

1. The input argument `m0` was renamed to `mInitial`.
2. The input argument `x0` was renamed to `xInitial`.
3. A new input parameter `mMinMax` controlling the minimum and maximum values of the restart parameter `m` can now be given by the user.
4. A new input parameter `mStep` controlling the step of the restart parameter in the `pd_rule` function can now also be given by the user.
5. The input variables `alphaP` and `alphaD` should now be given as a single input variable `alphaPD`, which is a `2-by-1` vector.

Default values and sanity checks were updated/added accordingly. Documentation on the input parameters was considerably improved. In addition, some cleaning of the code was performed.

In this PR, the test `test_pd_gmres` was renamed to `test_poisson` to better reflect the purpose of the tests.

### CODE REVIEW

File: src/pd_gmres.m [hunk 1]
```diff
@@ -97,32 +123,60 @@
...
+% ----> Default values and sanity checks for mMinMax
+if (nargin < 4) || isempty(mMinMax)
+    mMinMax = [1; n-1];
+else
+    if ~restarted
+        warning("mMinMax was given but will not be used.")
     end
 end
 
+% TODO: Do we want to allow for mMinMax(1) = mMinMax(2)? I guess in that
```
Comment (jhabriel) [inline src/pd_gmres.m:154]: @jccf19 Please have a look at this question.

File: src/pd_gmres.m [hunk 2]
```diff
@@ -97,32 +123,60 @@
...
+% ----> Default values and sanity checks for mStep
+if (nargin < 5) || isempty(mStep)
+    mStep = 1;
+else
+    if ~restarted
+        warnign("mStep was given but will not be used.")
+    end
+end
+   
+%TODO: Do we want to allow for mStep = 0? In that case, the m parameter
```
Comment (jhabriel) [inline src/pd_gmres.m:172]: @jccf19 Please have a look at this question.

File: src/pd_gmres.m (hunk 1 above)
Comment (jccf19) [reply src/pd_gmres.m:154]: Should be mMinMax(1) < mMinMax(2).

File: src/pd_gmres.m (hunk 2 above)
Comment (jccf19) [reply src/pd_gmres.m:172]: Should be  mStep>0.

**Comment by jhabriel** [comment]
I will close this PR, as PR #27 is a superset of this one.

**Pull request closed without merging**